#! /bin/sh /usr/share/dpatch/dpatch-run
## support-gzipped-charmaps.dpatch by Martin Pitt <martin.pitt@ubuntu.com>
##
## DP: Support gzipped charmaps.

@DPATCH@
diff -urNad cups-1.4.3~/cups/transcode.c cups-1.4.3/cups/transcode.c
--- cups-1.4.3~/cups/transcode.c	2009-05-14 23:18:35.000000000 +0200
+++ cups-1.4.3/cups/transcode.c	2010-05-21 15:39:34.462642308 +0200
@@ -1450,6 +1450,7 @@
   cups_ucs2_t	*crow;			/* Pointer to UCS-2 row in 'char2uni' */
   cups_sbcs_t	*srow;			/* Pointer to SBCS row in 'uni2char' */
   char		line[256];		/* Line from charset map file */
+  char		gzfilename[1027];	/* Filename with .gz suffix */
 
 
  /*
@@ -1476,9 +1477,14 @@
 
   if ((fp = cupsFileOpen(filename, "r")) == NULL)
   {
-    DEBUG_printf(("8get_sbcs_charmap: Returning NULL (%s)", strerror(errno)));
+    snprintf (gzfilename, sizeof(gzfilename), "%s.gz", filename);
 
-    return (NULL);
+    if ((fp = cupsFileOpen(gzfilename, "r")) == NULL)
+    {
+      DEBUG_printf(("8get_sbcs_charmap: Returning NULL (%s)", strerror(errno)));
+
+      return (NULL);
+    }
   }
 
  /*
@@ -1602,6 +1608,7 @@
   char		line[256];		/* Line from charset map file */
   int		i;			/* Loop variable */
   int		legacy;			/* 32-bit legacy char */
+  char		gzfilename[1027];	/* Filename with .gz suffix */
 
 
   DEBUG_printf(("7get_vbcs_charmap(encoding=%d, filename=\"%s\")\n",
@@ -1628,9 +1635,14 @@
 
   if ((fp = cupsFileOpen(filename, "r")) == NULL)
   {
-    DEBUG_printf(("8get_vbcs_charmap: Returning NULL (%s)", strerror(errno)));
+    snprintf (gzfilename, sizeof(gzfilename), "%s.gz", filename);
 
-    return (NULL);
+    if ((fp = cupsFileOpen(gzfilename, "r")) == NULL)
+    {
+      DEBUG_printf(("8get_vbcs_charmap: Returning NULL (%s)", strerror(errno)));
+
+      return (NULL);
+    }
   }
 
  /*
